www/js/app.js

```diff
@@ -580,17 +580,49 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
      * Displays the zone to select files from the archive
      */
     function displayFileSelect() {
-        $('#openLocalFiles').show();
-        $('#archiveFiles').on('change', setLocalArchiveFromFileSelect);
+        document.getElementById('openLocalFiles').style.display = 'block';
+        // Make the whole app a drop zone
+        var dropZone = document.getElementById('search-article');
+        dropZone.addEventListener('dragover', function(e) {
+            e.stopPropagation();
+            e.preventDefault();
+            e.dataTransfer.dropEffect = 'copy';
+        }, false);
+        dropZone.addEventListener('drop', setLocalArchiveFromFileList);
+        document.getElementById('archiveFiles').addEventListener('change', setLocalArchiveFromFileSelect);
     }
 
     function setLocalArchiveFromFileList(files) {
+        // Check to see if the files are from file select or from drag-and-drop
```

Comment:
Maybe it would more clear to create a dedicated function like `setLocalArchiveFromDragAndDroppedFiles`, that takes the files from the dataTransfer, and then calls `setLocalArchiveFromFileList`

Reply:
OK, will see what I can do.

Reply:
Done, I hope, but please review in due course.